src/components/UpupUploader/PreviewComponent.tsx

```diff
@@ -0,0 +1,86 @@
+import { bytesToSize } from 'lib'
+import {
+    HTMLAttributes,
+    forwardRef,
+    memo,
+    useEffect,
+    useMemo,
+    useRef,
+} from 'react'
+import { TbX } from 'react-icons/tb'
+import { FileHandlerProps, FileWithId } from 'types/file'
+import FilePreview from '../FilePreview'
+
+type Props = {
+    setFiles: FileHandlerProps['setFiles']
+    multiple?: boolean
+    mini?: boolean
+    file: FileWithId
+    index: number
+} & HTMLAttributes<HTMLDivElement>
+
+export default memo(
+    forwardRef<HTMLDivElement, Props>(function PreviewComponent(
+        { setFiles, multiple, mini, file, index, ...restProps }: Props,
+        ref,
+    ) {
+        const objectUrls = useRef<string[]>([])
+
+        useEffect(() => {
+            return () => {
+                objectUrls.current.forEach(url => URL.revokeObjectURL(url))
+                objectUrls.current = []
+            }
+        }, [])
+
+        const objectUrl = useMemo(() => {
+            const url = URL.createObjectURL(file)
+            objectUrls.current.push(url)
+            return url
+        }, [file])
```

Comment:
This could lead the url being added multiple times -> memory leaks if the useMemo is triggered multiple times (can happen alot with javascript.)

make sure the URL isn't already in the list before adding it

Reply:
Thanks @MedAmine1212, done